# HG changeset patch
# User rk
# Date 1286195720 -7200
# Node ID cfb42f3f572179d8eddac76f36797a279c3b0dba
# Parent  7489a3b6fb109cc9cc3b56e1660de1ba28e9a650
avoid making copies of nodes and children arrays when we only need them as const references

diff -r 7489a3b6fb10 -r cfb42f3f5721 src/generic/datavgen.cpp
--- a/src/generic/datavgen.cpp	Fri Oct 01 11:42:01 2010 +0200
+++ b/src/generic/datavgen.cpp	Mon Oct 04 14:35:20 2010 +0200
@@ -1896,8 +1896,8 @@ bool Walker( wxDataViewTreeNode * node, 
             ;
     }
 
-    wxDataViewTreeNodes nodes = node->GetNodes();
-    wxDataViewTreeLeaves leaves = node->GetChildren();
+    const wxDataViewTreeNodes& nodes = node->GetNodes();
+    const wxDataViewTreeLeaves& leaves = node->GetChildren();
 
     int len_nodes = nodes.GetCount();
     int len = leaves.GetCount();
